// @posthog/core/dist/process/cli.mjs
import { spawnLocal } from "./spawn-local.mjs";
function buildSourcemapCliArgs(config, mode) {
    const args = [
        'sourcemap',
        'process'
    ];
    if ('stdin' in mode) args.push('--stdin');
    else args.push('--directory', mode.directory);
    if (config.sourcemaps.releaseName) args.push('--release-name', config.sourcemaps.releaseName);
    if (config.sourcemaps.releaseVersion) args.push('--release-version', config.sourcemaps.releaseVersion);
    if (config.sourcemaps.deleteAfterUpload) args.push('--delete-after');
    if (config.sourcemaps.batchSize) args.push('--batch-size', config.sourcemaps.batchSize.toString());
    return args;
}
function buildCliEnv(config) {
    return {
        ...process.env,
        RUST_LOG: `posthog_cli=${config.logLevel}`,
        POSTHOG_CLI_HOST: config.host,
        POSTHOG_CLI_API_KEY: config.personalApiKey,
        POSTHOG_CLI_PROJECT_ID: config.projectId
    };
}
async function runSourcemapCli(config, options) {
    const mode = 'filePaths' in options ? {
        stdin: true
    } : {
        directory: options.directory
    };
    const args = buildSourcemapCliArgs(config, mode);
    const env = buildCliEnv(config);
    const spawnOptions = {
        cwd: process.cwd(),
        env,
        stdio: 'inherit'
    };
    if ('filePaths' in options) spawnOptions.stdin = options.filePaths.join('\n') + '\n';
    await spawnLocal(config.cliBinaryPath, args, spawnOptions);
}
export { buildCliEnv, buildSourcemapCliArgs, runSourcemapCli };
