// @posthog/core/dist/process/config.mjs
import { resolveBinaryPath } from "./utils.mjs";
function resolveConfig(options, resolveOptions) {
    const projectId = options.projectId ?? options.envId;
    const host = options.host ?? 'https://us.i.posthog.com';
    const logLevel = options.logLevel ?? 'info';
    const cwd = resolveOptions?.cwd ?? process.cwd();
    const cliBinaryPath = options.cliBinaryPath ?? resolveBinaryPath('posthog-cli', {
        path: process.env.PATH ?? '',
        cwd
    });
    const userSourcemaps = options.sourcemaps ?? {};
    const defaultEnabled = resolveOptions?.defaultEnabled ?? true;
    const enabled = userSourcemaps.enabled ?? defaultEnabled;
    if (enabled) {
        if (!projectId) throw new Error('projectId is required when sourcemaps are enabled (envId is deprecated)');
        if (!options.personalApiKey) throw new Error('personalApiKey is required when sourcemaps are enabled');
    }
    return {
        personalApiKey: options.personalApiKey,
        projectId,
        host,
        logLevel,
        cliBinaryPath,
        sourcemaps: {
            enabled,
            releaseName: userSourcemaps.releaseName ?? userSourcemaps.project,
            releaseVersion: userSourcemaps.releaseVersion ?? userSourcemaps.version,
            deleteAfterUpload: userSourcemaps.deleteAfterUpload ?? true,
            batchSize: userSourcemaps.batchSize
        }
    };
}
export { resolveConfig };
